File: AlumniDarli/app/Models/Comment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Comment extends Model
{
    use HasFactory;

    protected $table = 'comments';
    protected $fillable = [
        'user_id',
        'target_user_id',
        'content',
        'rating',
        'parent_id',
        'admin_status',
        'mudir_status',
        'admin_reply',
        'admin_reply_date',
        'mudir_reply',
        'mudir_reply_date'
    ];

    // Relasi: user yang memberi komentar
    public function user()
    {
        return $this->belongsTo(User::class, 'user_id', 'id_user');
    }

    // Relasi: user yang dikomentari
    public function target()
    {
        return $this->belongsTo(User::class, 'target_user_id', 'id_user');
    }

    // Relasi: balasan dari komentar
    public function replies()
    {
        return $this->hasMany(Comment::class, 'parent_id', 'id');
    }

    // Relasi: komentar induk (jika ini balasan)
    public function parent()
    {
        return $this->belongsTo(Comment::class, 'parent_id', 'id');
    }


}

File: Darli/app/src/main/java/com/example/darli/data/model/Comment.kt
package com.example.darli.data.model

data class Comment(
    val id: Int,
    val user_id: Int,
    val user_name: String,
    val user_photo: String?,
    val content: String,
    val rating: Int,
    val status: String,
    val admin_reply: String?,
    val admin_reply_date: String?,
    val mudir_reply: String?,
    val mudir_reply_date: String?,
    val created_at: String,
    val replies: List<CommentReply> = emptyList()
)

data class CommentReply(
    val user_name: String,
    val content: String,
    val created_at: String
)

data class CommentListResponse(
    val response_code: Int,
    val content: List<Comment>
)

data class StoreCommentResponse(
    val response_code: Int,
    val message: String
)
